rescript/tests/test_get_unite.py

```diff
@@ -0,0 +1,98 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2019-2023, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import pkg_resources
+import tempfile
+import pandas.core.frame
+import q2_types.feature_data
+from qiime2.plugin.testing import TestPluginBase
+from rescript.get_unite import (
+    UNITE_DOIS,
+    _unite_get_url,
+    _unite_get_tgz,
+    _unite_get_artifacts,
+    get_unite_data,
+)
+
+from urllib.request import urlopen
+from unittest.mock import patch
+
+
+class TestGetUNITE(TestPluginBase):
+    package = "rescript.tests"
+
+    def setUp(self):
+        super().setUp()
+        self.unitefile = pkg_resources.resource_filename(
+            "rescript.tests", "data/unitefile.tgz"
+        )
+        self.unitefile_no_dev = pkg_resources.resource_filename(
+            "rescript.tests", "data/unitefile_no_dev.tgz"
+        )
+
+    # Requires internet access
+    def test_unite_get_url(self):
+        # for all combinations...
+        for v in UNITE_DOIS.keys():
+            for tg in UNITE_DOIS[v].keys():
+                for s in UNITE_DOIS[v][tg].keys():
+                    # ... try to get the URL
+                    url = _unite_get_url(v, tg, s)
+                    urlopen(url)
+        self.assertTrue(True)
+
+    # Requires internet access
+    def test_unite_get_tgz(self):
+        # Download a single, small, unrelated file for testing
+        url = "https://files.plutof.ut.ee/doi/C9/F6/C9F687C997F72F674AA539CB80BF5D5BF6D1F402A2ACF840B20322850D3DFBA4.zip"  # noqa E501
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            # Test working URL
+            _unite_get_tgz(url, tmpdirname)
+            # Test bad url
+            with self.assertRaises(ValueError):
```

Comment:
a few different value errors are possible. Could you use [`assertRaisesRegex`](https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertRaisesRegex) for a more specific test?

Reply:
In my current setup, only one error is thrown: "File incomplete on try ?"

Interinally, the `response` is a 404 HTTP error, but I don't handle that differently

Would you like me to check for various HTTP status codes?
(I considered that, but the most common error I found in testing was an incomplete file, so I focused on that)

Reply:
no you do not need to check for specific HTTP status codes, as long as these are all caught by the one ValueError. There should be a test for each error or warning message raised in your code, but we do not need to be more granular than that. My point was just that you should check that the correct error message is being delivered (so test that the code is failing in the right spot)